### PR TITLE
fix: add functionality to bump a version

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

This is to handle bumping a version of a delegation https://github.com/sigstore/root-signing/pull/410#issuecomment-1261097991 after https://github.com/sigstore/root-signing/pull/327

### CODE REVIEW

File: pkg/repo/repo.go [hunk 1]
```diff
@@ -302,3 +303,55 @@ func UpdateRoleKeys(repo *tuf.Repo, store tuf.LocalStore, role string, keys []*d
...
+	if err != nil {
+		return err
+	}
+	return store.SetMeta(role, b)
+}
+
+// BumpMetadataVersion increments the version of the manifest.
+// This ONLY handles targets types! The repo.SetRootVersion, repo.SetTargetsVersion,
+// or repo.SetSnapshotVersion, or repo.SetTimestampVersion handle top-level
+// metadata.
```
Comment (joshuagl) [inline pkg/repo/repo.go]:
```suggestion
// BumpMetadataVersion increments the version of the manifest.
// This ONLY handles delegated targets roles! The repo.SetRootVersion,
// repo.SetTargetsVersion, or repo.SetSnapshotVersion, or
// repo.SetTimestampVersion handle top-level metadata.
```

We should provide better API for this in go-tuf :-|

File: pkg/repo/repo.go [hunk 2]
```diff
@@ -302,3 +303,55 @@ func UpdateRoleKeys(repo *tuf.Repo, store tuf.LocalStore, role string, keys []*d
...
+		return err
+	}
+	targets := &data.Targets{}
+	if err := json.Unmarshal(s.Signed, targets); err != nil {
+		return err
+	}
+	targets.Version++
+
+	signed, err := MarshalMetadata(targets)
+	if err != nil {
```
Comment (joshuagl) [inline pkg/repo/repo.go:353]: ooi why not:
```suggestion
	signed, err := MarshalMetadata(targets); err != nil {
```
?

**Review by joshuagl — DISMISSED** [review]
Minor comments, but looks fine to me.

File: pkg/repo/repo.go (hunk 1 above)
Comment (joshuagl) [reply pkg/repo/repo.go]: Filed an issue in go-tuf https://github.com/theupdateframework/go-tuf/issues/406

**Comment by asraa** [comment]
TODO: I just realized this may not refresh the expiration. This isn't strictly needed right now (because the particular file I need to refresh expires in January), but I will add a comment mentioning that in the doc string so we don't forget later.

File: pkg/repo/repo.go (hunk 2 above)
Comment (asraa) [reply pkg/repo/repo.go:353]: Hmmm I don't think that works: I need the result `signed` in a later line, and if I inline that, then `signed` is a local variable to the if scope.

**Review by bobcallaway — APPROVED** [review]

File: pkg/repo/repo.go (hunk 2 above)
Comment (joshuagl) [reply pkg/repo/repo.go:353]: That's me not grokking Go, thanks for taking the time to explain!

**Review by joshuagl — APPROVED** [review]

**Pull request merged**